nfs_v3_config_test.go

```diff
@@ -0,0 +1,274 @@
+package nfsv3driver_test
+
+import (
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/lager/lagertest"
+	"code.cloudfoundry.org/nfsv3driver"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("BrokerConfigDetails", func() {
+	var (
+		logger lager.Logger
+	)
+
+	BeforeEach(func() {
+		logger = lagertest.NewTestLogger("test-broker-config")
+	})
+
+	Context("ConfigDetails", func() {
+		It("returns empty mount & source options, no mandatory", func() {
+			ConfigMandatory := []string{}
+
+			configDetails := nfsv3driver.NewNfsV3ConfigDetails()
+			configDetails.ReadConf("", "", ConfigMandatory)
+
+			logger.Debug("configDetails Debug", lager.Data{"configDetails": *configDetails})
+
+			Expect(len(configDetails.Allowed)).To(Equal(0))
+			Expect(len(configDetails.Forced)).To(Equal(0))
+			Expect(len(configDetails.Options)).To(Equal(0))
+			Expect(configDetails.IsSloppyMount()).To(BeFalse())
+			Expect(len(configDetails.CheckMandatory())).To(Equal(0))
+		})
+
+		It("returns empty mount & source options, with mandatory", func() {
+			ConfigMandatory := []string{"uid", "gid"}
+			configDetails := nfsv3driver.NewNfsV3ConfigDetails()
+			configDetails.ReadConf("", "", ConfigMandatory)
+
+			logger.Debug("configDetails Debug", lager.Data{"configDetails": *configDetails})
+
+			Expect(len(configDetails.Allowed)).To(Equal(0))
+			Expect(len(configDetails.Forced)).To(Equal(0))
+			Expect(len(configDetails.Options)).To(Equal(0))
+			Expect(configDetails.IsSloppyMount()).To(BeFalse())
+			Expect(len(configDetails.CheckMandatory())).To(Equal(2))
+			Expect(configDetails.CheckMandatory()).To(Equal(ConfigMandatory))
+		})
+
+		It("returns not empty allowed & default, with mandatory", func() {
```

Comment:
wording needs rework please consider "given mandatory and allowed config, returns forced... " ?